internal/storage/bucket_handle.go

```diff
@@ -549,9 +549,58 @@ func (bh *bucketHandle) DeleteFolder(ctx context.Context, folderName string) (er
 	return err
 }
 
+func isPreconditionFailed(err error) (bool, error) {
```

Comment:
you can write unit tests for this method.

Reply:
Thanks for suggestion. added.